### PR TITLE
OCPNODE-1343: Selinux Workaround

### PR DESCRIPTION
**Pull request by rphillips**
This PR adds support for applying the work-around for selinux relabelling. To enable the functionality a few things are needed to be setup.

Label
```
forceselinuxrelabel.admission.node.openshift.io/enabled: "true"
```

## Deploy operator
```bash
export LOCAL_OPERATOR_IMAGE=quay.io/ryan.phillips/cluster-resource-override-operator:dev
export LOCAL_OPERAND_IMAGE=quay.io/ryan.phillips/cluster-resource-override:dev
make deploy
```

## Steps to test

Create the namespace with the selinux fix label:
```
oc create -f artifacts/example/test-namespace.yaml
```

Enable the operator boolean to turn on the selinux fix
```
oc create -f artifacts/example/selinux-fix-enable.yaml
```

Add the SCC changes to allow for svc_t labels:
```
oc create -f artifacts/example/test-selinux-scc.yaml
oc adm policy add-scc-to-user custom -z default -n {namespace}
``` 

Create a pod with a PersistentVolume (PV):
```
oc create -f artifacts/example/test-selinux-pod.yaml
```

## Dependent PRs
https://github.com/openshift/cluster-resource-override-admission/pull/37

### CODE REVIEW

File: pkg/apis/autoscaling/v1/override_types.go [hunk 1]
```diff
@@ -128,6 +128,9 @@ type PodResourceOverrideSpec struct {
 	// LimitRange defaults are merged prior to the override.
 	//
 
+	// EnableSelinuxFix (if true) label pods with svc_t if they have a PVC
```
Comment (jrvaldes) [inline pkg/apis/autoscaling/v1/override_types.go:131]: https://github.com/openshift/cluster-resource-override-admission/pull/37#discussion_r1044745952

File: pkg/generated/clientset/internalversion/clientset.go [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2021 Red Hat, Inc.
+Copyright 2022 Red Hat, Inc.
```
Comment (jrvaldes) [inline pkg/generated/clientset/internalversion/clientset.go:2]: Consider adding to the commit message the command used to generate the files.

File: pkg/asset/daemonset.go [hunk 3]
```diff
@@ -70,7 +70,7 @@ func (d *daemonset) New() *appsv1.DaemonSet {
 								"--bind-address=127.0.0.1",
 								"--tls-cert-file=/var/serving-cert/tls.crt",
 								"--tls-private-key-file=/var/serving-cert/tls.key",
-								"--v=3",
+								"--v=5",
```
Comment (jrvaldes) [inline pkg/asset/daemonset.go:73]: If this change is not related to the Selinux workaround, consider moving to a separate PR to bump the default verbosity of the operand.

**Review by jrvaldes — COMMENTED** [review]
@rphillips Thanks for working on this. PTAL at my comments.

File: pkg/generated/clientset/internalversion/clientset.go (hunk 2 above)
Comment (rphillips) [reply pkg/generated/clientset/internalversion/clientset.go:2]: this is generated code.

File: pkg/asset/daemonset.go (hunk 3 above)
Comment (rphillips) [reply pkg/asset/daemonset.go]: fixed!

File: pkg/apis/autoscaling/v1/override_types.go (hunk 1 above)
Comment (rphillips) [reply pkg/apis/autoscaling/v1/override_types.go]: fixed!

**Comment by jrvaldes** [comment]
LGTM

@rphillips the verify job [failed](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-resource-override-admission-operator/95/pull-ci-openshift-cluster-resource-override-admission-operator-master-verify/1625593862636441600).

**Comment by jrvaldes** [comment]
@rphillips  the last test [passed](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-resource-override-admission-operator/95/pull-ci-openshift-cluster-resource-override-admission-operator-master-e2e-aws-operator/1625992463221002240#1:build-log.txt%3A50) and failed at deprovision.

**Comment by rphillips** [comment]
@jrvaldes thanks for checking... Talked to Trevor in slack... He's ok overriding if that is the case.

/override ci/prow/e2e-aws-operator
